pkg/controllers/cassandra/role/control_test.go

```diff
@@ -1,40 +1,93 @@
 package role_test
 
 import (
-	"fmt"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/jetstack/navigator/internal/test/unit/framework"
+	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+	"github.com/jetstack/navigator/pkg/controllers"
 	"github.com/jetstack/navigator/pkg/controllers/cassandra/role"
 	casstesting "github.com/jetstack/navigator/pkg/controllers/cassandra/testing"
 )
 
 func TestRoleSync(t *testing.T) {
-	t.Run(
-		"role missing",
-		func(t *testing.T) {
-			f := casstesting.NewFixture(t)
-			f.Run()
-			f.AssertRolesLength(1)
+	cluster1 := casstesting.ClusterForTest()
+	role1 := role.RoleForCluster(cluster1)
+
+	type testT struct {
+		kubeObjects        []runtime.Object
+		cluster            *v1alpha1.CassandraCluster
+		fixtureManipulator func(*testing.T, *framework.StateFixture)
```

Comment:
FYI, I introduced something similar in cert-manager: https://github.com/jetstack/cert-manager/pull/309/files#diff-db7d66ec1a20f75e1ea91f197b370598R27

Mine introduces a `PreFn` and ` CheckFn` which is passed the fixture structure so it can be manipulated/prepared 😄